src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -0,0 +1,198 @@
+    // Add{Thread|WiFi}Network and EnableNetwork have higher priority than other commands, implement them first.
+}
+
+// TODO: We cannot get length of ssid / credentials here, should be fixed after we have byte string type in ZCL codegen.
+// Note: SSID and credentials are not necessarily to be string.
+void OnAddWiFiNetworkCommandCallback(app::Command *, EndpointId, const uint8_t * ssid, const uint8_t * credentials,
+                                     uint64_t breadcrumb, uint32_t timeoutMs)
+{
+    CHIP_ERROR err = CHIP_ERROR_NO_MEMORY;
+
+    for (size_t i = 0; i < kMaxNetworks; i++)
+    {
+        if (sNetworks[i].mNetworkType == NetworkType::kUndefined)
+        {
+            // Assume they are null terminated, see notes above.
+            size_t ssid_Len        = strlen(reinterpret_cast<const char *>(ssid));
+            size_t credentials_Len = strlen(reinterpret_cast<const char *>(credentials));
+            memcpy(sNetworks[i].mData.mWiFi.mSSID, ssid, ssid_Len);
+            sNetworks[i].mData.mWiFi.mSSIDLen = ssid_Len;
+            memcpy(sNetworks[i].mData.mWiFi.mCredentials, credentials, credentials_Len);
+            sNetworks[i].mData.mWiFi.mCredentialsLen = credentials_Len;
+            memcpy(sNetworks[i].mNetworkID, sNetworks[i].mData.mWiFi.mSSID, ssid_Len);
+            sNetworks[i].mNetworkIDLen = ssid_Len;
+
+            sNetworks[i].mNetworkType = NetworkType::kWiFi;
+            sNetworks[i].mEnabled     = false;
+
+            err = CHIP_NO_ERROR;
+            break;
+        }
+    }
+
+    SuccessOrExit(err);
+
+    ChipLogDetail(Zcl, "WiFi provisioning data: SSID: %s", ssid);
+exit:
+    // TODO: We should encode response command here.
+    ChipLogFunctError(err);
+}
+
+namespace {
+CHIP_ERROR DoEnableNetwork(NetworkInfo & network)
+{
+    switch (network.mNetworkType)
+    {
+    case NetworkType::kThread:
+        // TODO(#5040): Add support for thread network provisioning.
+        return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+    case NetworkType::kWiFi:
+#if defined(CHIP_DEVICE_LAYER_TARGET)
+    {
+        DeviceLayer::DeviceNetworkProvisioningDelegateImpl deviceDelegate;
+        return deviceDelegate.ProvisionWiFi(reinterpret_cast<const char *>(network.mData.mWiFi.mSSID),
+                                            reinterpret_cast<const char *>(network.mData.mWiFi.mCredentials));
+    }
+#else
+        return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE);
+#endif
+    break;
+    case NetworkType::kEthernet:
+    case NetworkType::kUndefined:
+    default:
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+}
+} // namespace
+
+void OnEnableNetworkCommandCallback(app::Command *, EndpointId, const uint8_t * networkID, uint64_t breadcrumb, uint32_t timeoutMs)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    size_t networkSeq;
+    size_t networkID_Len = strlen(reinterpret_cast<const char *>(networkID));
+
+    err = CHIP_ERROR_KEY_NOT_FOUND;
```

Comment:
Maybe `CHIP_ERROR err = CHIP_ERROR_KEY_NOT_FOUND` instead of setting the value later

Reply:
Made a change for changing all functions in this file "Internal", and then made them return the enum type for errors.